networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/beaconchain/methods/BlobSidecarsByRangeMessageHandler.java

```diff
@@ -203,50 +227,76 @@ class RequestState {
     private final UInt64 maxRequestBlobSidecars;
     private final UInt64 startSlot;
     private final UInt64 endSlot;
+    private final UInt64 finalizedSlot;
+    private final Map<UInt64, Bytes32> canonicalHotRoots;
 
-    private Optional<Iterator<SlotAndBlockRootAndBlobIndex>> iterator = Optional.empty();
+    private Optional<Iterator<SlotAndBlockRootAndBlobIndex>> blobSidecarKeysIterator =
```

Comment:
Maybe add a comment, something like includes both canonical and non-canonical blob sidecars on a hot part?